src/CribblyBackend.DataAccess/Repositories/TeamRepository.cs

```diff
@@ -21,7 +22,32 @@ public TeamRepository(IDbConnection connection)
         {
             _connection = connection;
         }
+        public async Task<List<Team>> GetAll()
+        {
+            var players = new List<Player>();
+            var teams = (await _connection.QueryAsync<Team, Player, Team>(
+                @"SELECT * FROM Teams t INNER JOIN Players p ON t.Id = p.TeamId",
+                (t, p) => 
+                {
+                    switch (players.Count)
+                    {
+                        case 0:
+                            players.Add(p);
+                            break;
+                        case 1:
+                            players.Add(p);
+                            t.Players = new List<Player>();
+                            t.Players.AddRange(players);
+                            players.Clear();
+                            break;
+                    }
 
+                    return t;
+                },
+                splitOn: "Id"
+                )).Where(t => t.Players != null).ToList();
```

Comment:
Do we need this `Where`? `INNER JOIN` will not return any teams if there are no players for that team

Reply:
The SQL as is returns 4 records in my database with 2 teams. Two of the records have the teams and two of the records have players as a null property. It is kind of fucky, but I am all ears if there is a better way to map the properties

Reply:
Refactoring to move mapping outside of the anonymous function would remove the need for the ```WHERE``` statement now that I think about it

Reply:
Okay - maybe I'm confusing myself. I thought `INNER JOIN` joins two rows given a common ID (in this case joining on the team's ID and the player's TeamId foreign key), and would return no results if there were no players that had the given team ID?

Reply:
That's all correct. The thing is that there are two records returned per team, because there are two players associated with each team, hence the 4 records returned in my case. I will try using a dictionary, which should hopefully remove the need for this LINQ

Reply:
ah, ok, then I assume in your two results per team, everything related to the team is the exact same in both of those rows? I think the dictionary may help... we'll see :sunglasses:

Reply:
This code is now gone!